app/components/post_list/post_list.js

```diff
@@ -248,16 +248,31 @@ export default class PostList extends PureComponent {
     };
 
     renderItem = ({item, index}) => {
+        const {
+            highlightPinnedOrFlagged,
+            highlightPostId,
+            isSearchResult,
+            lastPostIndex,
+            location,
+            onHashtagPress,
+            onPostPress,
+            postIds,
+            renderReplies,
+            shouldRenderReplyButton,
+            theme,
+        } = this.props;
+
         if (PostListUtils.isStartOfNewMessages(item)) {
             // postIds includes a date item after the new message indicator so 2
             // needs to be added to the index for the length check to be correct.
-            const moreNewMessages = this.props.postIds.length === index + 2;
+            const moreNewMessages = postIds.length === index + 2;
+            const checkForPostId = index < postIds.length - 3;
```

Comment:
please add a comment explaining why the magic `- 3`

Reply:
The date line and new message line each count for a line. So the goal of this is to check for the 3rd previous, which for the start of a thread would be null as it doesn't exist.

Reply:
Cool, can you add the comment in the code? Thx.